File: JAVA/MbtiPersonalityTest.java
import java.util.Scanner;
public class MbtiPersonalityTest{
	static Scanner input = new Scanner(System.in);

	public static void main(String[] args){

		System.out.println("""
				MBTI Personality Test 
""");
		
		String judgingPerceiving = getJudgingPerceiving();
		String extraversionIntroversion = getExtraversionIntroversion();
		String thinkingFeeling = getThinkingFeeling();
		String sensingIntuition = getSensingIntuition();
		String adaptability = getAdaptability();
		String personalityType = sensingIntuition + extraversionIntroversion + thinkingFeeling + judgingPerceiving + adaptability;

		System.out.println("\nYour personality type is: " + personalityType);
    }

	public static String getJudgingPerceiving(){
		System.out.println("\nWhen working on a project, what motivates you more?");
		System.out.println("A. Breaking down the task into smaller, manageable parts and creating a schedule.");
		System.out.println("B. Brainstorming new ideas and exploring different possibilities.");
		System.out.print("\nEnter your choice (A/B): ");
		String choice = input.next();
		return choice.equals("A") ? "J" : "P";


    }

	public static String getExtraversionIntroversion(){
		System.out.println("\nIn social situations, how do you typically feel?");
		System.out.println("A. Energized and enthusiastic, enjoying the company of others.");
		System.out.println("B. Reserved and reflective, preferring to observe before contributing.");
		System.out.print("\nEnter your choice (A/B): ");
		String choice = input.next().toUpperCase();
		return choice.equals("A") ? "E" : "I";
    }

	public static String getThinkingFeeling(){
		System.out.println("\nWhen making decisions, what do you prioritize?");
		System.out.println("A. Logical analysis and objective criteria.");
		System.out.println("B. Personal values and the impact on others.");
		System.out.print("\nEnter your choice (A/B): ");
        	String choice = input.next().toUpperCase();
		return choice.equals("A") ? "T" : "F";
    }
	public static String getSensingIntuition(){
		System.out.println("\nWhen solving problems, what approach do you typically take?");
		System.out.println("A. Focusing on concrete facts and details.");
		System.out.println("B. Exploring abstract ideas and patterns.");
		System.out.println("\nEnter your choice (A/B)");
		String choice = input.next().toUpperCase();
		return choice.equals("A") ? "S" : "I";
 }

	public static String getAdaptability(){
		System.out.println("How do you handle unexpected changes in plans?");
		System.out.println("A. Adapting quickly and flexibly.");
		System.out.println("B. Preferring to stick to the original plan.");
		System.out.print("\nEnter your choice (A/B): ");
		String choice = input.next().toUpperCase();
		return choice.equals("A") ? "P" : "J";



}
}

File: PYTHON/mbtipersonalitytest.py
print("""
		MBTI PERSONALITY TEST
""")

print("mbtipersonalitytest __name__:", __name__)


def main():
    judging_perceiving = get_judging_perceiving()
    extraversion_introversion = get_extraversion_introversion()
    thinking_feeling = get_thinking_feeling()
    sensing_intuition = get_sensing_intuitioSn()
    personality_type = sensing_intuition + extraversion_introversion + thinking_feeling + judging_perceiving
    print("\nYour personality type is:", personality_type)

def get_judging_perceiving():
    print("\nWhen working on a project, what motivates you more?")
    print("A. Breaking down the task into smaller, manageable parts and creating a schedule.")
    print("B. Brainstorming new ideas and exploring different possibilities.")
    choice = input("\nEnter your choice (A/B): ").upper()
    if choice == "A":
        return "J"
    else:
        return "P"

def get_extraversion_introversion():
    print("\nIn social situations, how do you typically feel?")
    print("A. Energized and enthusiastic, enjoying the company of others.")
    print("B. Reserved and reflective, preferring to observe before contributing.")
    choice = input("\nEnter your choice (A/B): ").upper()
    if choice == "A":
        return "E"
    else:
        return "I"

def get_thinking_feeling():
    print("\nWhen making decisions, what do you prioritize?")
    print("A. Logical analysis and objective criteria.")
    print("B. Personal values and the impact on others.")
    choice = input("\nEnter your choice (A/B): ").upper()
    if choice == "A":
        return "T"
    else:
        return "F"

def get_sensing_intuition():
    print("\nWhen solving problems, what approach do you typically take?")
    print("A. Focusing on concrete facts and details.")
    print("B. Exploring abstract ideas and patterns.")
    choice = input("\nEnter your choice (A/B): ").upper()
    if choice == "A":
        return "S"
    else:
        return "N" 

def get_adaptability():
    print("\nHow do you handle unexpected changes in plans?")
    print("A. Adapting quickly and flexibly.")
    print("B. Preferring to stick to the original plan.")
    choice = input("\nEnter your choice (A/B): ").upper()
    if choice == "A":
        return "P"
    else:
        return "J"

if __name__ == "__main__":
    main()
